examples/c/event/src/listener.c

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#include "iox2/iceoryx2.h"
+
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+int main(void) {
+    // create new node
+    iox2_node_builder_h node_builder_handle = iox2_node_builder_new(NULL);
+    iox2_node_h node_handle = NULL;
+    if (iox2_node_builder_create(node_builder_handle, NULL, iox2_service_type_e_IPC, &node_handle) != IOX2_OK) {
+        printf("Could not create node!\n");
+        goto end;
+    }
+
+    // create service name
+    const char* service_name_value = "MyEventName";
+    iox2_service_name_h service_name = NULL;
+    if (iox2_service_name_new(NULL, service_name_value, strlen(service_name_value), &service_name) != IOX2_OK) {
+        printf("Unable to create service name!\n");
+        goto drop_node;
+    }
+
+    // create service
+    iox2_service_name_ptr service_name_ptr = iox2_cast_service_name_ptr(service_name);
+    iox2_node_ref_h node_ref_handle = iox2_cast_node_ref_h(node_handle);
+    iox2_service_builder_h service_builder = iox2_node_service_builder(node_ref_handle, NULL, service_name_ptr);
+    iox2_service_builder_event_h service_builder_event = iox2_service_builder_event(service_builder);
+    iox2_port_factory_event_h service = NULL;
+    if (iox2_service_builder_event_open_or_create(service_builder_event, NULL, &service) != IOX2_OK) {
+        printf("Unable to create service!\n");
+        goto drop_node;
+    }
```

Comment:
I think for this we could create a convenience API like `iox2_event_service_create_with_default_settings(char*, iox2_port_factory_event_h*)`. Not for this PR but in the short term to make the examples less intimidating.
Alternatively, instead of following the Rust builder too closely, we could also create a struct with all the settings and pass that to a function.

What do you think?

Reply:
I like it! This API requires the user really to write a lot.